DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+# How to Create and Review PRs
+
+## Overview
+
+Blockchain software development requires a much higher degree of rigor than most other kinds of software.  This is because with blockchains, **there is no roll-back** from a bad deployment.  There is essentially zero room for consensus bugs.  If you ship a consensus bug, that bug could not only have catastrophic consequences for users (i.e. they lose all their money), but also be intractable to fix, mitigate, or remove.  This is because unlike nearly every other kind of networked software, **the state of the blockchain is what the users' computers say it is.**  If you want to make changes, you _must_ get _user_ buy-in, and this is necessarily time-consuming and not at all guaranteed to succeed.
+
+Therefore, making changes to the codebase is necessarily a review-intensive process.  No one wants bugs, but **no one can afford consensus bugs**.  This page describes how to make and review _non-consensus_ changes.  The process for consensus changes includes not only the entirety of this document, but also the [SIP process](https://github.com/stacksgov/sips/blob/main/sips/sip-000/sip-000-stacks-improvement-proposal-process.md).
+
+A good PR review sets both the submitter and reviewers up for success.  It minimizes the time required by both parties to get the code into an acceptable state, without sacrificing quality or safety.  Unlike most other software development practices, _safety_ is the primary concern.  A PR can and will be delayed or closed if there is any concern that it will lead to unintended consensus-breaking changes.
+
+This document describes some best practices on how to create and review PRs in this context.  The target audience is people who have commit access to this repository (reviewers), and people who open PRs (submitters).  This is a living document -- developers can and should document their own additional guidelines here.
```

Comment:
Should we be more inclusive of devs who don't have access as well? For example, devs who fork the repo and open a PR against upstream? Can we create guidelines for how these folks can also contribute?

Reply:
I think the language makes sense as-is. in this section, the target audience is:
1. reviewers who have commit access to the repo
2. submitters who are opening PR's (i.e. from a personal fork)

I think there is a case to be made about *what* to do in the case of 2 (i.e. merge from upstream/develop before opening PR).

Reply:
I agree -- I'm going to mark this as resolved for now, because I think we can always expand the audience of this document later on.